migrations/handle_subscriptions.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ *
+ * phpBB Browser Push Notifications. An extension for the phpBB Forum Software package.
+ *
+ * @copyright (c) 2023, phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ */
+
+namespace phpbb\webpushnotifications\migrations;
+
+use phpbb\db\migration\migration;
+
+class handle_subscriptions extends migration
+{
+	public function effectively_installed()
+	{
+		return !$this->db_tools->sql_table_exists($this->table_prefix . 'wpn_notification_push');
+	}
+
+	public function revert_data(): array
+	{
+		return [
+			['custom', [[$this, 'update_subscriptions']]],
+			['if', [
+				($this->db_tools->sql_table_exists($this->table_prefix . 'notification_push')),
+				['custom', [[$this, 'copy_subscription_tables']]],
+			]],
+			['if', [
+				(isset($this->config['webpush_enable'])),
+				['config.update', ['webpush_enable', $this->config['wpn_webpush_enable']]],
+			]],
+			['if', [
+				(isset($this->config['webpush_vapid_public'])),
+				['config.update', ['webpush_vapid_public', $this->config['wpn_webpush_vapid_public']]],
+			]],
+			['if', [
+				(isset($this->config['webpush_vapid_private'])),
+				['config.update', ['webpush_vapid_private', $this->config['wpn_webpush_vapid_private']]],
+			]],
+		];
+	}
+
+	public function update_subscriptions()
```

Comment:
Lets just add some docblock comments to this and the function below so we can remember what they are for/doing in the future.